### PR TITLE
Linux kernel updates 2023-07-05

### PR DESCRIPTION
**Pull request by fabianhjr**
###### Description of changes

Previous: https://github.com/NixOS/nixpkgs/pull/241655 (originally as https://github.com/NixOS/nixpkgs/pull/240987 )

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabianhjr** [comment]
Only built and tested 6.4.2

**Review by Ma27 — CHANGES_REQUESTED** [review]
Fails for me with

```
error: build of '/nix/store/pz9ha43a433sqdhn3zx9l8dmy16cr09i-linux-6.1.38.tar.xz.drv' on 'ssh-ng://tum-graham' failed: hash mismatch in fixed-output derivation '/nix/store/pz9ha43a433sqdhn3zx9l8dmy16cr09i-linux-6.1.38.tar.xz.drv':
         specified: sha256-RsrXEtJhojyOSDo7ebaoS5pfcxqJIckSffNa41zvHoA=
            got:    sha256-+aT5G2CffTMqXyvgGrhjNvoAFJ+ua9wZ8W+hn3iALUM=
copying path '/nix/store/0zzdkalc9crn0knpakckz0sasb8qzblx-linux-hardened-5.10.186-hardened1.patch' from 'ssh-ng://tum-graham'
error: builder for '/nix/store/pz9ha43a433sqdhn3zx9l8dmy16cr09i-linux-6.1.38.tar.xz.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/zxdnadb2m94hg49wmd5qilkj8q3qk3d0-linux-6.1.38.drv' failed to build
error: 1 dependencies of derivation '/nix/store/l4yj5bvzhhaz0jndr7a2q9qbldr1dz2j-linux-6.1.38-modules.drv' failed to build
linux-config> building '/nix/store/35bijchbhr6sjyi3p3a5i3sqih8k3gwv-linux-config-5.10.186.drv' on 'ssh-ng://tum-graham'
error: 1 dependencies of derivation '/nix/store/4x2hk5kwfavng43jb3x11sjdqnhjri9i-nixos-system-machine-23.11pre-git.drv' failed to build
error: 1 dependencies of derivation '/nix/store/59k5aq5k3skrzr6yc6875ph7dq6priw7-nixos-vm.drv' failed to build
error: 1 dependencies of derivation '/nix/store/gzrvkz7dyjq2wa72z39v0mvsb8cf4a0s-nixos-test-driver-kernel-6.1.38.drv' failed to build
```

**Comment by fabianhjr** [comment]
:O oh no, looking into it

File: pkgs/os-specific/linux/kernel/linux-6.1.nix
Comment (fabianhjr) [inline pkgs/os-specific/linux/kernel/linux-6.1.nix:1]: Indeed, hash wasn't updated

Comment (fabianhjr) [reply pkgs/os-specific/linux/kernel/linux-6.1.nix:1]: Attempted to run the update script a second time and same result.

Manually using nix-prefetch-url gives:

```
diff --git a/pkgs/os-specific/linux/kernel/linux-6.1.nix b/pkgs/os-specific/linux/kernel/linux-6.1.nix
index aed17f49b94..2b6dfd3e145 100644
--- a/pkgs/os-specific/linux/kernel/linux-6.1.nix
+++ b/pkgs/os-specific/linux/kernel/linux-6.1.nix
@@ -13,6 +13,6 @@ buildLinux (args // rec {
 
   src = fetchurl {
     url = "mirror://kernel/linux/kernel/v6.x/linux-${version}.tar.xz";
-    hash = "sha256-RsrXEtJhojyOSDo7ebaoS5pfcxqJIckSffNa41zvHoA=";
+    sha256 = "0hrdh1w9z8bgy4cxqsxfkwa01yincfw1mq1bbwm36zczc0dzk97r";
   };
 } // (args.argsOverride or { }))
```

Comment (fabianhjr) [reply pkgs/os-specific/linux/kernel/linux-6.1.nix:1]: Reran the script amending from `hash` to `sha256` and this time that update successfully.

**Comment by cmm** [comment]
maybe this could be an opportunity to instead change `update.sh` to accept either `hash` or `sha256` and canonicalize whatever it was to `hash`.

**Comment by fabianhjr** [comment]
@cmm probably best on a separate merge request.

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 241759` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.kernels.linux_5_15</li>
    <li>linuxKernel.kernels.linux_5_15.dev</li>
    <li>linuxKernel.kernels.linux_5_15_hardened</li>
    <li>linuxKernel.kernels.linux_5_15_hardened.dev</li>
    <li>linuxKernel.kernels.linux_rt_5_4</li>
    <li>linuxKernel.kernels.linux_rt_5_4.dev</li>
  </ul>
</details>
<details>
  <summary>28 packages built:</summary>
  <ul>
    <li>linuxKernel.kernels.linux_4_14_hardened</li>
    <li>linuxKernel.kernels.linux_4_14_hardened.dev</li>
    <li>linuxKernel.kernels.linux_4_19_hardened</li>
    <li>linuxKernel.kernels.linux_4_19_hardened.dev</li>
    <li>linuxKernel.kernels.linux_5_10_hardened</li>
    <li>linuxKernel.kernels.linux_5_10_hardened.dev</li>
    <li>linuxKernel.kernels.linux_5_4_hardened</li>
    <li>linuxKernel.kernels.linux_5_4_hardened.dev</li>
    <li>linuxKernel.kernels.linux_6_1</li>
    <li>linuxKernel.kernels.linux_6_1.dev</li>
    <li>linuxKernel.kernels.linux_hardened (linuxKernel.kernels.linux_6_1_hardened)</li>
    <li>linuxKernel.kernels.linux_hardened.dev (linuxKernel.kernels.linux_6_1_hardened.dev)</li>
    <li>linuxKernel.kernels.linux_6_3</li>
    <li>linuxKernel.kernels.linux_6_3.dev</li>
    <li>linuxKernel.kernels.linux_6_3_hardened</li>
    <li>linuxKernel.kernels.linux_6_3_hardened.dev</li>
    <li>linuxKernel.kernels.linux_6_4 (linuxKernel.kernels.linux_testing)</li>
    <li>linuxKernel.kernels.linux_6_4.dev (linuxKernel.kernels.linux_testing.dev)</li>
    <li>linuxKernel.kernels.linux_latest_libre</li>
    <li>linuxKernel.kernels.linux_latest_libre.dev</li>
    <li>linuxKernel.kernels.linux_libre</li>
    <li>linuxKernel.kernels.linux_libre.dev</li>
    <li>linuxKernel.kernels.linux_rt_5_10</li>
    <li>linuxKernel.kernels.linux_rt_5_10.dev</li>
    <li>linuxKernel.kernels.linux_rt_5_15</li>
    <li>linuxKernel.kernels.linux_rt_5_15.dev</li>
    <li>linuxKernel.kernels.linux_testing_bcachefs</li>
    <li>linuxKernel.kernels.linux_testing_bcachefs.dev</li>
  </ul>
</details>

**Comment by Ma27** [comment]
Currently running tests, would merge tonight if everything goes well.

**Comment by Ma27** [comment]
OK since you don't allow maintainers to push on PRed branches, please remove the `fix-amdgpu-5_15` patch yourself.

**Comment by Ma27** [comment]
Rest of the updates seem OK, though.

**Comment by fabianhjr** [comment]
Oh, apoligies will do.

**Comment by fabianhjr** [comment]
Pushing seems to be related to the fork being on an org instead of my personal account; seems like the only workaround would be to give push permission to NixOS maintainers.

**Review by Ma27 — APPROVED** [review]

**Comment by vcunat** [comment]
> linux-rt_5_4: 5.4.242-rt81 -> 5.4.248-rt83

The build broke on Hydra, simultaneously with the version update:
https://hydra.nixos.org/build/226948220

(also on 23.05)

**Comment by Ma27** [comment]
Thx for pointing that out, fixing...

**Pull request merged**